qiskit/circuit/classicalfunction/classicalfunction.py

```diff
@@ -96,6 +94,7 @@ def types(self):
             ret.append({k: v[0] for k, v in scope.items()})
         return ret
 
+    @HAS_TWEEDLEDUM.require_in_call
```

Comment:
It's probably best to stick make this `require_in_instance` as well tbh - you can't do anything meaningful with the class at all without tweedledum.